tests/std/tests/P1682R3_to_underlying/test.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright(c) Microsoft Corporation.
+// SPDX - License - Identifier : Apache - 2.0 WITH LLVM - exception
+
+#include <cassert>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+// TRANSITION, GH-602
+// template <class T>
+// constexpr bool is_valid = requires(T e) { to_underlying(e); };
+
+template <class T, class = void>
+struct can_underlying : false_type {};
+template <class T>
+struct can_underlying<T, void_t<decltype(to_underlying(declval<T>()))>> : true_type {};
+
+template <class T>
+constexpr bool is_valid = can_underlying<T>::value;

```

Comment:
```suggestion
template <class, class = void>
constexpr bool is_valid = false;

template <class T>
constexpr bool is_valid<T, void_t<decltype(to_underlying(declval<T>()))>> = true;
```
(I might keep the more descriptive name `can_underlying` instead of the generic `is_valid`, but that's pure style so fine to keep as is.)

(I suppose you could also change this to `declval<T&>()` if you wanted to more closely mimic what the _requires-expression_ in the comment is testing.)